From 66103f2311e78a71d502d81b9ed54ef9867d162c Mon Sep 17 00:00:00 2001
From: Manigandan Gopalakrishnan <manigandan.gopalakrishnan@ltts.com>
Date: Sat, 13 Aug 2022 23:34:53 +0530
Subject: [PATCH] REFPLTB-1741: Running dibbler-server with brlan0 in reference
 platform target

Reason for change: To run dibbler server with brlan0 interface
Test procedure: Verified that connected client got global IPv6 address
Risks: None

Signed-off-by: Manigandan Gopalakrishnan <manigandan.gopalakrishnan@ltts.com>
---
 recipes-ccsp/util/utopia.bbappend            | 10 ++++++++++
 recipes-ccsp/util/utopia/dibbler-client.conf | 17 +++++++++++++++++
 recipes-ccsp/util/utopia/dibbler-server.conf | 10 ++++++++++
 3 files changed, 37 insertions(+)
 create mode 100644 recipes-ccsp/util/utopia/dibbler-client.conf
 create mode 100644 recipes-ccsp/util/utopia/dibbler-server.conf

diff --git a/recipes-ccsp/util/utopia.bbappend b/recipes-ccsp/util/utopia.bbappend
index 6010fc8..becb2b5 100644
--- a/recipes-ccsp/util/utopia.bbappend
+++ b/recipes-ccsp/util/utopia.bbappend
@@ -18,6 +18,8 @@ SRC_URI += "file://posix-gwprovapp.patch;apply=no"
 SRC_URI += "file://0002-fix-swctl-missing-api.patch;apply=no"
 SRC_URI += "file://start_dnsmasq_service.patch;apply=no"
 SRC_URI += "file://dhcp_script.sh"
+SRC_URI += "file://dibbler-server.conf"
+SRC_URI += "file://dibbler-client.conf"
 
 LDFLAGS_append = " \
     -lsecure_wrapper \
@@ -172,6 +174,14 @@ do_install_append() {
 
     echo "sysevent set bridge_mode \`syscfg get bridge_mode\`" >> ${D}${sysconfdir}/utopia/utopia_init.sh
     echo "sysevent set lan-status started" >> ${D}${sysconfdir}/utopia/utopia_init.sh
+
+#IPv6 support: starting dibbler-server
+    install -d ${D}${sysconfdir}/dibbler
+    install -m 0644 ${WORKDIR}/dibber-server.conf ${D}${sysconfdir}/dibbler/server.conf
+    install -m 0644 ${WORKDIR}/dibber-client.conf ${D}${sysconfdir}/dibbler/client.conf
+    echo "echo_t '[utopia][init] Triggering ipv6-start'" >> ${D}${sysconfdir}/utopia/utopia_init.sh
+    echo "/etc/utopia/service.d/service_ipv6.sh ipv6-start &" >> ${D}${sysconfdir}/utopia/utopia_init.sh
+
     echo 'echo_t "[utopia][init] completed creating utopia_inited flag"' >> ${D}${sysconfdir}/utopia/utopia_init.sh
 
 #WanManager Feature
diff --git a/recipes-ccsp/util/utopia/dibbler-client.conf b/recipes-ccsp/util/utopia/dibbler-client.conf
new file mode 100644
index 0000000..e0f43f0
--- /dev/null
+++ b/recipes-ccsp/util/utopia/dibbler-client.conf
@@ -0,0 +1,17 @@
+script "/lib/rdk/client-notify.sh"
+log-level 7
+log-mode full
+duid-type duid-ll
+iface erouter0 {
+        ia
+        pd
+{ prefix ::/56 }
+
+        option dns-server
+        option domain
+
+        option 0016 hex 0x0000118b000a65526f75746572312e30
+        option 0017 hex 0x0000118b0002000745524f555445520003001045434d3a454456413a45524f555445520026
+}
+skip-confirm
+downlink-prefix-ifaces "brlan0"
diff --git a/recipes-ccsp/util/utopia/dibbler-server.conf b/recipes-ccsp/util/utopia/dibbler-server.conf
new file mode 100644
index 0000000..7ffe364
--- /dev/null
+++ b/recipes-ccsp/util/utopia/dibbler-server.conf
@@ -0,0 +1,10 @@
+log-level 6
+iface brlan0 {
+  preference 7
+  class {
+    pool 2601:9c0:200:2300:0:0:0:0001 - 2601:9c0:200:2300:0:0:0:fffe
+    prefered-lifetime 226688
+    valid-lifetime 226688
+  }
+option dns-server 2001:558:feed::1,2001:558:feed::2
+}
-- 
2.17.1

